### PR TITLE
Update composer dependencies and allow minor version updates to Twig

### PR DESCRIPTION
**Pull request by Ayesh**
Twig recently had a security vulnerability (https://symfony.com/blog/twig-security-release-possibility-to-load-a-template-outside-a-configured-directory-when-using-the-filesystem-loader), and our `composer.json` version constraint required one of the vulnerable versions.

This commit relaxes the `twig/twig` dependency to allow minor upgrades, and makes routine updates to other Composer dependencies as well.

### CODE REVIEW

File: .github/workflows/deploy-site.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     - name: Setup PHP
       uses: shivammathur/setup-php@v2
       with:
-        php-version: 8.0
+        php-version: '8.1'
```
Comment (sebastianbergmann) [inline .github/workflows/deploy-site.yml:30]: This is not necessary and unrelated to "Update composer dependencies and allow minor version updates to Twig".

File: .github/workflows/netlify.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 8.0
+          php-version: '8.1'
```
Comment (sebastianbergmann) [inline .github/workflows/netlify.yml:35]: This is not necessary and unrelated to "Update composer dependencies and allow minor version updates to Twig".

File: .github/workflows/netlify.yml [hunk 3]
```diff
@@ -55,4 +55,4 @@ jobs:
         env:
           NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
           NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
-        timeout-minutes: 1
\ No newline at end of file
+        timeout-minutes: 1
```
Comment (sebastianbergmann) [inline .github/workflows/netlify.yml:58]: This looks like a superfluous whitespace change.

**Comment by Ayesh** [comment]
Thank you @sebastianbergmann for the pointers, this indeed was a messy PR. I was having trouble with the CI was using PHP 8.0 instead PHP 8.1 in order for Composer to properly resolve the dependency list. I will close this PR for now, and make a more organized one.

**Pull request closed without merging**